icinga2/icinga.go

```diff
@@ -141,6 +161,40 @@ func (s *WebClient) FilteredQuery(url string, filter QueryFilter, result, errmsg
 	return s.napping.Send(&req)
 }
 
+func (s *WebClient) SetIcingaUrl(url string) {
+	s.URL = url
+}
+
+func (s *MockClient) SetIcingaUrl(url string) {
+	conf := s.GetClientConfig()
+	conf.SetIcingaUrl(url)
+}
```

Comment:
Mock implementations should never call the type they're mocking. The simplest solution here is to add field `URL` to `MockClient`.

```suggestion
func (s *MockClient) SetIcingaUrl(url string) {
	s.URL = url
}
```